### PR TITLE
Gc mock package

### PR DESCRIPTION
**Pull request by dadams39**
## Description
Command line utility that allows for the printing of mockData based on the m365ID. Supports:
- `exchange.Mail` 
- `exchange.Contacts`
- `exchange.Events`

## Type of change

- [x] :sunflower: Feature

## Issue(s)

Related to:
* #1255<issue>
* #999 
* #705

### CODE REVIEW

File: src/cmd/mckgen/mckgen.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	RunE:  handleMockCommand,
+}
+
+// Required inputs from user from the user
+var (
+	tenant, user, m365ID, category string
+)
+
+// main function will produce the json String for a given m365 object of a signed
+// in user. Displayed Objects can be used as inputs for Mockable data
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/cmd/mckgen/mckgen.go]: Resolved with [796a8da](https://github.com/alcionai/corso/pull/1313/commits/796a8da9f1c468636a81baee9a680b511c6c57a0)

File: src/cmd/mckgen/mckgen.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+		if err != nil {
+			return errors.Wrapf(err, "unable to %s to string value", m365ID)
+		}
+
+		array, err := sw.GetSerializedContent()
+		if err != nil {
+			return errors.Wrapf(err, "unable to serialize new value from M365:%s", m365ID)
+		}
+
+		fmt.Printf("%s\n", string(array))
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/cmd/mckgen/mckgen.go]: Resolved [796a8da](https://github.com/alcionai/corso/pull/1313/commits/796a8da9f1c468636a81baee9a680b511c6c57a0)

File: src/cmd/mckgen/mckgen.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+	sw := kw.NewJsonSerializationWriter()
+
+	response, err := get(ctx, gs, user, m365ID)
+	if err != nil {
+		return errors.Wrap(err, support.ConnectorStackErrorTrace(err))
+	}
+
+	// First return is the number of bytes that were serialized. Ignored
+	_, err = serializeFunc(ctx, gs.Client(), sw, channel, response, user)
+	close(channel)
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/cmd/mckgen/mckgen.go]: I moved this, but I am not sure if this is the correct application [796a8da](https://github.com/alcionai/corso/pull/1313/commits/796a8da9f1c468636a81baee9a680b511c6c57a0)

**Review by ryanfkeepers — APPROVED** [review]
Approved, since nothing here is blocking, but would prefer to see the package name change (mock -> getItem) reflected in the command handling (which is still using command and explanations for "mock").  The identity is a little disjointed right now.

File: src/cmd/mckgen/mckgen.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dadams39) [reply src/cmd/mckgen/mckgen.go]: Resolved with [df72bbe](https://github.com/alcionai/corso/pull/1313/commits/df72bbeb632996b807a5719e5d20e2a1768f1330)

**Pull request merged**